### ISSUE TITLE
wrong typespec for Task.Supervisor.start_link

### ISSUE THREAD

**Issue by disya2** [elixir-lang/elixir#6023]
I get dialyzer errors when I use Task.Supervisor.start_link with `:restart` and/or `:shutdown` options. I believe that's because those two options are not in `Supervisor.options`.

### Environment
```
Erlang/OTP 19 [erts-8.3] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]
Elixir 1.4.2
```

MacOS X and Ubuntu Linux 14 and 16

### Current behavior
The example code 
```
defmodule X do
  def start(_, _) do
    Task.Supervisor.start_link([restart: :transient])
  end
end
```
when run through dialyzer produces the following error:
```
lib/x.ex:2: Function start/2 has no local return
lib/x.ex:3: The call 'Elixir.Task.Supervisor':start_link([{'restart','transient'},...]) breaks the contract ('Elixir.Supervisor':options()) -> 'Elixir.Supervisor':on_start()
```
### Expected behavior
Dialyzer should not produce any errors.

**Comment by ericmj** [comment]
The restart and shutdown options are on the child specification, the supervisor should not have these options.

Supervisor options: http://erlang.org/doc/man/supervisor.html#type-sup_flags
Child options: http://erlang.org/doc/man/supervisor.html#type-child_spec

**Comment by fishcakez** [comment]
Closed in 4360198bbfb5a81441c8c5d1275181e3909f9262.
